src/services/worker_task_tracking_repository.js

```diff
@@ -7,25 +7,52 @@ This Source Code Form is subject to the terms of the Mozilla Public License, v.
 This Source Code Form is “Incompatible With Secondary Licenses”, as defined by the Mozilla Public License, v. 2.0.
 */
 
-
 export default class WorkerTaskTrackingRepository {
   constructor(db) {
     this.db = db;
   }
 
-  async tryToBeginWork(workType) {
-    try {
-      const {insertedId} = await this.db.collection('workerTasks').insertOne({workType});
-      return insertedId;
-    } catch (error) {
-      if (error.message.includes(`duplicate key error`)) {
-        throw new Error('Work of this type is currently in progress');
-      }
-      throw new Error(error);
+  statuses = {
+    running: 'RUNNING',
+    timeout: 'TIMEOUT',
+    success: 'SUCCESS',
+    failure: 'FAILURE'
```

Comment:
I'm not sure if we can assume the worker to be as kind as to let us know about failure in performing the work. As we're doing this bottom-up (we don't have usages of `finishWork` yet), I'd postpone this until later and now stick to just 3 statuses: running, timeout, finished.